Do not pass -fstack-protector unconditionally

Using -fstack-protector only works when the toolchain has SSP support.

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>

Index: b/src/Makefile.am
===================================================================
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -127,7 +127,6 @@
 	-Wformat \
 	-Wshadow \
 	-fno-omit-frame-pointer \
-	-fstack-protector \
 	$(GLIB2_CFLAGS) \
 	$(LIBNL_CFLAGS) \
 	$(NCURSES_CFLAGS) \
